### PR TITLE
new module: kaiju/mkfmi

### LINKED ISSUE #2955: New module: kaiju/mkbwt and kaiju/mkfmi
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by alxndrdiaz**
## PR checklist

Closes #2955 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by alxndrdiaz** [comment]
All local tests work just fine (conda, docker and singularity) using: `nf-core modules create-test-yml -t kaiju/mkfmi`  But, they fail when running from GitHub actions.

**Comment by jfy133** [comment]
Awesome! Thanks for taking this over @alxndrdiaz !

I am going to be traveling today so won't be able to review today, however I will be able to tomorrow. But if you want this reviewed faster it'll make sense to ask others for review.

@sofstam might be a good person for example :)

> All local tests work just fine (conda, docker and singularity) using: `nf-core modules create-test-yml -t kaiju/mkfmi`  but they fail when running from GitHub actions. 

I believe there has been a hiccup last night a broken PR that got merged. Try pulling master into your branch later today and the tests will run properly I expect 👍

File: modules/nf-core/kaiju/mkfmi/meta.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: "kaiju_mkfmi"
+description: Make Kaiju FMI-index file from a protein FASTA file
+keywords:
+  - classify
+  - metagenomics
+  - fastq
+  - taxonomic profiling
```
Comment (jfy133) [inline modules/nf-core/kaiju/mkfmi/meta.yml:7]:
```suggestion
  - taxonomic profiling
  - database
  - index
```

File: modules/nf-core/kaiju/mkfmi/meta.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      licence: ["GNU GPL v3"]
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'test', single_end:false ]`
+  - fasta:
+      type: file
+      description: Protein FASTA file
```
Comment (jfy133) [inline modules/nf-core/kaiju/mkfmi/meta.yml:24]:
```suggestion
      description: Uncompressed Protein FASTA file (mandatory)
```

Possibly optional, given my comment above

Adding the mandatory is per https://nf-co.re/docs/contributing/modules#documentation point 12

**Review by jfy133 — APPROVED** [review]
A few minor things but otherwise LGTM

File: modules/nf-core/kaiju/mkfmi/main.nf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    kaiju-mkbwt \\
+        $args \\
+        -n $task.cpus \\
+        -o ${prefix} \\
+        ${fasta}
```
[the remark this answers is not in the document]
Comment (alxndrdiaz) [reply modules/nf-core/kaiju/mkfmi/main.nf:28]: > Should we include a conditional step to insert a command unzipping the FASTA iif it's given as zipped as per guidelines [1](#user-content-fn-1-e6e598e83d77ab3e597665e05e637e28)
> ## Footnotes
> 
>     1. https://nf-co.re/docs/contributing/modules#general point 7 [↩](#user-content-fnref-1-e6e598e83d77ab3e597665e05e637e28)

Committed all suggestions, except this ones. So the uncompressed FASTA is mandatory.

**Pull request merged**